p2p/discovery/mdns.go

```diff
@@ -165,7 +138,7 @@ func (m *mdnsService) handleEntry(e *mdns.ServiceEntry) {
 	}
 
 	maddr, err := manet.FromNetAddr(&net.TCPAddr{
-		IP:   e.AddrV4,
+		IP:   e.AddrIPv4[0],
```

Comment:
As there is possibility of multiple addresses, I would iterate over them, crate maddr for each and add it to the peerstore.